addons/test_mail/tests/test_mail_mail.py

```diff
@@ -823,6 +830,97 @@ def test_mail_mail_values_unicode(self):
+                ((1, self.env['ir.mail_server'], 0.3, cst_max_size_always_exceed, True), False),
+                # 3 attachment: exceed max size
+                ((3, self.env['ir.mail_server'], 0.3, cst_max_size_never_exceed, True), True),
+                # 1 attachment: exceed max size
+                ((1, self.env['ir.mail_server'], 0.1, cst_max_size_never_exceed, True), True),
+                # Same as above with a specific server. Note that the default and server max_email size are reversed.
+                ((1, test_mail_server, cst_max_size_always_exceed, 0.3, True), False),
+                ((3, test_mail_server, cst_max_size_never_exceed, 0.3, True), True),
+                ((1, test_mail_server, cst_max_size_never_exceed, 0.1, True), True),
+                # Attachments not linked to a business record are never turned to link
+                ((3, self.env['ir.mail_server'], 0.3, cst_max_size_never_exceed, False), False),
+                ((1, test_mail_server, cst_max_size_never_exceed, 0.1, False), False),
+        ):
+            self.env['ir.config_parameter'].sudo().set_param('base.default_max_email_size', default_email_size)
+            if mail_server:
+                mail_server.max_email_size = max_email_size
+            attachments = self.env['ir.attachment'].sudo().create([{
+                'name': f'attachment{idx_attachment}',
+                'res_name': 'test',
+                'res_model': self.test_record._name if business_attachment else 'mail.message',
+                'res_id': self.test_record.id if business_attachment else 0,
+                'datas': datas,
+            } for idx_attachment in range(n_attachment)])
+            with self.mock_smtplib_connection():
+                mails = self.env['mail.mail'].create([{
+                    'attachment_ids': attachments.ids,
+                    'body_html': '<p>Test</p>',
+                    'email_from': 'test@test_2.com',
+                    'email_to': f'u_{mail_idx}_s-{bool(mail_server)}_a-{n_attachment}_l-{expected_is_links}@test.com',
+                } for mail_idx in range(2)])
+                mails._send(mail_server=mail_server)
+
+            match_download_links = [
+                re.compile(fr'.*<a.*href.*/web/content/([^?]*)\?.*access_token.*>.*{attachment.name}.*</a>', re.DOTALL)
+                for attachment in attachments]
+            self.assertEqual(len(self.emails), 2)
+            for mail, outgoing_email in zip(mails, self.emails):
+                message_raw = outgoing_email['message']
+                message_parsed = message_from_string(message_raw)
+                message_cleaned = re.sub(r'[\s=]', '', message_raw)
+                with self.subTest(email_to=mail['email_to']):
+                    if expected_is_links:
+                        self.assertEqual(count_attachments(message_parsed), 0,
+                                         'Attachment have been removed (replaced by download links)')
```

Comment:
Attachments*

As this message is printed when this does NOT happen, would change "have been" to "should have been removed and replaced by".

To apply to other messages as well.